search-app/src/cmr/search/api/concepts_search.clj

```diff
@@ -252,6 +252,32 @@
                   (pr-str params)))
     (core-api/search-response ctx (query-svc/get-deleted-granules ctx params))))
 
+(defn- validate-content-type
+  "Validate the given headers has the expected content type."
+  [headers expected-content-type]
+  (let [content-type-header (get headers (string/lower-case common-routes/CONTENT_TYPE_HEADER))]
+    (when-not (= expected-content-type content-type-header)
+      (svc-errors/throw-service-error
+       :invalid-content-type (str "Unsupported content type [" content-type-header "]")))))
+
+(defn- clear-scroll
+  "Invokes query service to clear the scroll context for the given scroll id.
+   The concept type of the request must be JSON."
+  [context headers body]
+  (validate-content-type headers mt/json)
+  (if-let [short-scroll-id (-> body
+                               (json/parse-string true)
+                               :scroll_id)]
+    (if-let [scroll-id (->> short-scroll-id
+                            (core-api/get-scroll-id-and-search-params-from-cache context)
+                            :scroll-id)]
+      (query-svc/clear-scroll context scroll-id)
+      (svc-errors/throw-service-error
+       :invalid-data (format "scroll_id [%s] not found." short-scroll-id)))
```

Comment:
Having a nested `if-let` makes this hard to read. Is there a way for you to break this into a separate function or make `scroll-id` and `short-scroll-id` a part of the same `let` block?

Reply:
This function is effectively two `if-let` blocks. I don't know how to make it easier to read. :) Breaking it into more functions would be harder to follow in my opinion.

Reply:
Added comments. Hopefully, this makes the code easier to follow.